### PR TITLE
update mint price

### LINKED ISSUE #300: Update mint price before mint starts

### PR DESCRIPTION
**Pull request by yubrew**
fix #300

### CODE REVIEW

**Review by shanev — APPROVED** [review]
LGTM, just have to be updated.

File: contracts/vending-minter/src/contract.rs [hunk 1]
```diff
@@ -590,6 +592,45 @@ fn random_mintable_token_mapping(
...
+    }
+
+    // TODO add test
+    // If price is greater than initial price, return error
+    if price > config.extension.initial_price.amount.u128() {
+        return Err(ContractError::UpdatedMintPriceTooHigh {
+            allowed: config.extension.initial_price.amount.u128(),
+            updated: price,
+        });
+    }
```
Comment (jhernandezb) [inline contracts/vending-minter/src/contract.rs:624]: This won't let update price above the initial price even if the public start time hasn't started.

**Review by jhernandezb — CHANGES_REQUESTED** [review]

File: contracts/vending-minter/src/contract.rs (hunk 1 above)
Comment (yubrew) [reply contracts/vending-minter/src/contract.rs:624]: is there a point to `initial_price` if you can set it above or below before the sale and it's not used after sale?

File: contracts/vending-minter/src/contract.rs [hunk 2]
```diff
@@ -590,6 +591,36 @@ fn random_mintable_token_mapping(
...
+    price: u128,
+) -> Result<Response, ContractError> {
+    nonpayable(&info)?;
+    let mut config = CONFIG.load(deps.storage)?;
+    if info.sender != config.extension.admin {
+        return Err(ContractError::Unauthorized(
+            "Sender is not an admin".to_owned(),
+        ));
+    }
+    // TODO add test
```
Comment (shanev) [inline contracts/vending-minter/src/contract.rs:607]:
```suggestion
```

**Review by shanev — APPROVED** [review]

**Review by jhernandezb — APPROVED** [review]

**Pull request merged**